src/understand-tidb/privilege-management.md

```diff
@@ -1,2 +1,65 @@
+
+2. In the executor function which handles the statement (examples: [`SHOW PROCESSLIST`](https://github.com/pingcap/tidb/blob/1a54708a7f8f86515236626c78e97a33d8adf583/executor/show.go#L368-L380)).
+
+The first option is recommended, as it is much less verbose. However, `visitInfo` does not handle cases where the statement can behave differently depending on the permissions of the user executing it. All users can execute the `SHOW PROCESSLIST` statement, but to see the sessions of other users requires the `PROCESS` privilege.
+
+`visitInfo` also does not support the case that a statement requires either `XYZ` **or** `ZYX` as a permission. An example of this, is that dropping a user in MySQL requires either the `CREATE USER` privilege, or the `DELETE` privilege on the `mysql.user` table. Permissions can only be attached in `visitInfo` with an **AND** semantic.
+
+### Manually checking with the privilege manager
+
+For (2) above, manual checks should follow the following pattern:
+
+```go
+checker := privilege.GetPrivilegeManager(e.ctx)
+if checker != nil && !checker.RequestVerification(ctx.GetSessionVars().ActiveRoles, schema.Name.L, table.Name.L, "", mysql.AllPrivMask) {
+    /* do something */
+}
+..
+if checker == nil || !checker.RequestDynamicVerification(ctx.GetSessionVars().ActiveRoles, "RESTRICTED_TABLES_ADMIN", false) {
+    /* do something */
+}
+```
+
+The check for `checker != nil` is important because for internal SQL statements the privilege manager is not present. These statements are expected to fall through and satisfy the privilege check.
+
+### Static and dynamic privileges
+
+Privileges fall into two categories:
+* Static privileges: These are the "traditional" privileges such as `INSERT`, `UPDATE`, `SELECT`, `DELETE`, `SUPER`, `PROCESS` which have existed in MySQL for a long time. They can *usually* be assigned to a user on either a global or database/table level.
+* Dynamic privileges: These are new privileges such as `BACKUP_ADMIN`, `RESTORE_ADMIN`, `CONNECTION_ADMIN`. They can only be assigned on a global level, and each have their own "grantable" attribute.
+
+Dynamic privileges were introduced in MySQL 8.0 (and [TiDB 5.1](https://github.com/pingcap/tidb/blob/master/docs/design/2021-03-09-dynamic-privileges.md)) to solve a specific issue, which is that the `SUPER` privilege is too coarse. There are many scenarios where a user needs to be assigned the `SUPER` privilege to perform a specific action, but too many other privileges are granted at the same time.
+
+Typically requiring the `SUPER` privilege is an anti-pattern since almost all dynamic privileges are [also satisfied](https://github.com/pingcap/tidb/blob/5e05922de6a253859cfbfe19356de8a2e2db39da/privilege/privileges/cache.go#L1009) by the `SUPER` privilege.
```

Comment:
Please add a more clear message that no new features should require `SUPER` privileges, but a dynamic privilege, either existing or new.